dlt/destinations/impl/sink/configuration.py

```diff
@@ -0,0 +1,71 @@
+from typing import TYPE_CHECKING, Optional, Final, Callable, Union, Any
+from importlib import import_module
+
+from dlt.common.configuration import configspec
+from dlt.common.destination import TLoaderFileFormat
+from dlt.common.destination.reference import (
+    DestinationClientConfiguration,
+    CredentialsConfiguration,
+)
+from dlt.common.typing import TDataItems
+from dlt.common.schema import TTableSchema
+from dlt.common.configuration.exceptions import ConfigurationValueError
+
+
+TSinkCallable = Callable[[Union[TDataItems, str], TTableSchema], None]
+
+
+@configspec
+class SinkClientCredentials(CredentialsConfiguration):
```

Comment:
Those are not credentials. User will provide credentials (eventually) in the decorated function. if you need to keep the reference to decorated function, use some other field name

Reply:
Alright, moved the callable into the regular config